File: ros/src/test_py_pkg/test_py_pkg/test_publisher.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node

from example_interfaces.msg import String 

class Publisher(Node): # MODIFY NAME
    def __init__(self):
        super().__init__("test_publisher") # MODIFY NAME

        self.publisher_ = self.create_publisher(msg_type=String, topic="test_topic", qos_profile=10)
        self.time_ = self.create_timer(0.5, self.publish_topic)

        self.get_logger().info("Starting publisher...")

    def publish_topic(self):
        msg = String()
        msg.data = "Test"
        self.publisher_.publish(msg)
 
 
def main(args=None):
    rclpy.init(args=args)
    node = Publisher() # MODIFY NAME
    rclpy.spin(node)
    rclpy.shutdown()
 
 
if __name__ == "__main__":
    main()

File: ros/src/test_cpp_pkg/src/test_publisher.cpp
#include "example_interfaces/msg/string.hpp"
#include "rclcpp/rclcpp.hpp"

class test_publisher : public rclcpp::Node  // MODIFY NAME
{
   public:
    test_publisher() : Node("test_publisher"), name_("Björn")  // MODIFY NAME
    {
        publisher_ = this->create_publisher<example_interfaces::msg::String>("test_publisher", 10);
        timer_ = this->create_wall_timer(std::chrono::milliseconds(500), std::bind(&test_publisher::publish, this));

        RCLCPP_INFO(this->get_logger(), "cpp test_publisher has been started...");
    }

   private:
    void publish() {
        auto msg = example_interfaces::msg::String();
        msg.data = std::string("Hello, I'm ") + name_;
        publisher_->publish(msg);
    }

    std::string name_;
    rclcpp::Publisher<example_interfaces::msg::String>::SharedPtr publisher_;
    rclcpp::TimerBase::SharedPtr timer_;
};

int main(int argc, char **argv) {
    rclcpp::init(argc, argv);
    auto node = std::make_shared<test_publisher>();  // MODIFY NAME
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}